Datadog crash when attempting to update dashboard

Hey folks,

We're running into an issue with using Terraform to manage our Datadog dashboards. It looks like it might be similar to the issue seen in #1249 but I'm not seeing the same empty requests artifacts in our HCL.

### Terraform Version

```
1.1.6
```

(I know, not the latest version, but updating is a bit nontrivial. I'm happy to try if triage thinks that could legitimately solve the issue but reading this stack trace really looks like the plugin got something unexpected....)

### Affected Resource(s)

-   datadog_dashboard

### Terraform Configuration Files

Can't share the HCL in this forum, unfortunately, even if GPG encrypted.

### Debug Output

Can't share the Debug output in this forum, unfortunately, even if GPG encrypted.

### Panic Output

If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

```
Stack trace from the terraform-provider-datadog_v3.23.0 plugin:
panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 41 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogQueryValueRequests(0xc00e290568)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:3879 +0x8fa
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogQueryValueDefinition(0x1311460?)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:3687 +0xc5
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidget(0xc0001f23c0?)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:1058 +0xb6e
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidgets(0xc00e290948)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:975 +0xdb
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogDashboard(0xc00097c100)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:359 +0x4b9
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardCreate({0x17a0dc8, 0xc00092c000}, 0xc00097c100, {0x12206e0?, 0xc0005c0540})
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:148 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0001eed20, {0x17a0e00, 0xc00805d1a0}, 0xd?, {0x12206e0, 0xc0005c0540})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0001eed20, {0x17a0e00, 0xc00805d1a0}, 0xc00c3e3930, 0xc00d7af900, {0x12206e0, 0xc0005c0540})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000362000, {0x17a0e00?, 0xc00805d080?}, 0xc000122230)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000334e60, {0x17a0e00?, 0xc00805c840?}, 0xc000322070)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x14c1460?, 0xc000334e60}, {0x17a0e00, 0xc00805c840}, 0xc000322000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000003c0, {0x17a3b80, 0xc0004d6340}, 0xc008060000, 0xc00031e5d0, 0x2290620, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000003c0, {0x17a3b80, 0xc0004d6340}, 0xc008060000, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a
Error: The terraform-provider-datadog_v3.23.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

It should have created/updated the dashboard.

### Actual Behavior

Dashboard was unaffected.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

The dashboard is generated from customer config and is probably a good bit larger than the "average" dashboard.

### References

#1249

--- a/datadog/resource_datadog_dashboard.go
+++ b/datadog/resource_datadog_dashboard.go
@@ -1802,6 +1802,9 @@ func buildDatadogChangeRequests(terraformRequests *[]interface{}) *[]datadogV1.C
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogChangeRequest.SetFormulas(formulas)
@@ -3876,6 +3879,9 @@ func buildDatadogQueryValueRequests(terraformRequests *[]interface{}) *[]datadog
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogQueryValueRequest.SetFormulas(formulas)
@@ -4149,6 +4155,9 @@ func buildDatadogQueryTableRequests(terraformRequests *[]interface{}) *[]datadog
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogQueryTableRequest.SetFormulas(formulas)
@@ -5489,6 +5498,9 @@ func buildDatadogGeomapRequests(terraformRequests *[]interface{}) *[]datadogV1.G
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogGeomapRequest.SetFormulas(formulas)
@@ -5978,6 +5990,9 @@ func buildDatadogSunburstRequests(terraformRequests *[]interface{}) *[]datadogV1
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogSunburstRequest.SetFormulas(formulas)
@@ -6921,6 +6936,9 @@ func buildDatadogTimeseriesRequests(terraformRequests *[]interface{}) *[]datadog
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogTimeseriesRequest.SetFormulas(formulas)
@@ -7202,6 +7220,9 @@ func buildDatadogToplistRequests(terraformRequests *[]interface{}) *[]datadogV1.
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogToplistRequest.SetFormulas(formulas)
@@ -7605,6 +7626,9 @@ func buildDatadogTreemapRequests(terraformRequests *[]interface{}) *[]datadogV1.
 		if v, ok := terraformRequest["formula"].([]interface{}); ok && len(v) > 0 {
 			formulas := make([]datadogV1.WidgetFormula, len(v))
 			for i, formula := range v {
+				if formula == nil {
+					continue
+				}
 				formulas[i] = buildDatadogFormula(formula.(map[string]interface{}))
 			}
 			datadogTreemapRequest.SetFormulas(formulas)

